### PR TITLE
Get rid of `MemoryLoginsStorage.kt`

### LINKED ISSUE #2310: Get rid of `MemoryLoginsStorage.kt`
Ref this comment thread: https://github.com/mozilla/application-services/pull/2262/files#r351552656

We have an entirely separate implementation of the `LoginsStorage` API, implemented in Kotlin and backed by simple in-memory storage. It's a maintenance burden and IMHO is at substantial risk of diverging in behaviour from the canonical sqlite-backed version.

As far as I can tell, the only consumer of this implementation is Lockwise for Android, which uses it in tests. I recall some trouble with Lockwise using our rust-components in their tests at some point, but I can no longer find an open issue about it.

What would it take for us to replace this custom implementation with the rust one, perhaps backed by a `:memory:` sqlite database rather than a file on disk?

┆Issue is synchronized with this [Jira Story](https://jira.mozilla.com/browse/SYNC-471)

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #2310

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

**Comment by vladikoff** [comment]
@thomcc I have removed the interface. Do you have any preferences on how to restructure further:

image

Should we combine the 2 test files into one or keep them with different names?

File: components/logins/android/src/test/java/mozilla/appservices/logins/LoginsStorageTest.kt [hunk 1]
```diff
@@ -14,11 +14,11 @@ import org.junit.Assert.fail
 
 abstract class LoginsStorageTest {
```
Comment (thomcc) [inline components/logins/android/src/test/java/mozilla/appservices/logins/LoginsStorageTest.kt]: I think we should combine this and the other file too.

**Comment by eoger** [comment]
There’s an issue with this commit it includes Swift bits that are not supposed to be merged

File: components/fxa-client/ios/FxAClient/RustFxAccount.swift [hunk 2]
```diff
@@ -259,19 +266,28 @@ open class RustFxAccount {
         }
     }
 
-    open func migrateFromSessionToken(sessionToken: String, kSync: String, kXCS: String) throws -> Bool {
-        let json = try nullableRustCall { err in
+    open func migrateFromSessionToken(sessionToken: String, kSync: String, kXCS: String) throws -> String? {
+        let ptr_to_json = try nullableRustCall { err in
             fxa_migrate_from_session_token(self.raw, sessionToken, kSync, kXCS, 0 /* reuse session token */, err)
         }
-        // We don't parse the JSON coz nobody uses it...
```
[the remark this answers is not in the document]
Comment (vladikoff) [reply components/fxa-client/ios/FxAClient/RustFxAccount.swift]: @eoger should we fix this in another pr?

[the remark this answers is not in the document]
Comment (eoger) [reply components/fxa-client/ios/FxAClient/RustFxAccount.swift]: Yes please!

[the remark this answers is not in the document]
Comment (vladikoff) [reply components/fxa-client/ios/FxAClient/RustFxAccount.swift]: filed https://github.com/mozilla/application-services/issues/2835

**Pull request merged**